code/game/objects/structures/mineral_doors.dm

```diff
@@ -151,18 +157,30 @@
 	sheetType = /obj/item/stack/sheet/mineral/silver
 	max_integrity = 300
 
+/obj/structure/mineral_door/silver/Initialize()
+	. = ..()
+	AddComponent(/datum/component/rad_insulation, RAD_HEAVY_INSULATION)
+
 /obj/structure/mineral_door/gold
 	name = "gold door"
 	icon_state = "gold"
 	sheetType = /obj/item/stack/sheet/mineral/gold
 
+/obj/structure/mineral_door/gold/Initialize()
+	. = ..()
+	AddComponent(/datum/component/rad_insulation, RAD_HEAVY_INSULATION)
+
 /obj/structure/mineral_door/uranium
 	name = "uranium door"
 	icon_state = "uranium"
 	sheetType = /obj/item/stack/sheet/mineral/uranium
 	max_integrity = 300
 	light_range = 2
 
+/obj/structure/mineral_door/uranium/Initialize()
+	. = ..()
+	qdel(GetComponent(/datum/component/rad_insulation))

```

Comment:
Instead of deleting new components like this, Add a common proc that you just override for /uranium